### ISSUE TITLE
Possible to control EP600

### ISSUE THREAD

**Issue by smai86** [Patrick762/hassio-bluetti-bt#31]
Hey Patrick,

do you think that is possible to control the EP600 and change some values over HA? 
For me it would be perfekt to switch the "Main Switch" from the App to on/off. 

Greetings ausm Saarland ;-)
Steffen

![IMG_9626](https://github.com/Patrick762/hassio-bluetti-bt/assets/84518538/6b58ecb7-a416-4755-8504-0776b998ee42)

**Comment by Patrick762** [comment]
This would be a good thing as the EP600 discharges so much at night

**Comment by smai86** [comment]
Yes thats my problem... 
I build a switchbot with tasmota so i can use an automation in HA to press the "green button" on the EP600 to shutdown if my SOC is less than 5% and no sun in shining. And the other way the switchbot will press again to turn on the EP600 if there is pretty much sun that I load into grid. These things are working like charme, but I alway had to turn the "Main Switch" toggle in the App on manually. :-(

**Comment by smai86** [comment]
Yeah that would be awesome. Just let me know if i can help you out! Thank you for your work!

**Comment by carstenschroeder** [comment]
Would be great if this could be added! Is there anything I can do to help? I have a EP600 for testing.

**Comment by carstenschroeder** [comment]
> Yes thats my problem... I build a switchbot with tasmota so i can use an automation in HA to press the "green button" on the EP600 to shutdown if my SOC is less than 5% and no sun in shining. And the other way the switchbot will press again to turn on the EP600 if there is pretty much sun that I load into grid. These things are working like charme, but I alway had to turn the "Main Switch" toggle in the App on manually. :-(

@smai86 Did you have to open the B500 to connect the switchbot to the green button or how did you solve it? I would like to build something similar. :)

**Comment by smai86** [comment]
> > Yes thats my problem... I build a switchbot with tasmota so i can use an automation in HA to press the "green button" on the EP600 to shutdown if my SOC is less than 5% and no sun in shining. And the other way the switchbot will press again to turn on the EP600 if there is pretty much sun that I load into grid. These things are working like charme, but I alway had to turn the "Main Switch" toggle in the App on manually. :-(
> 
> @smai86 Did you have to open the B500 to connect the switchbot to the green button or how did you solve it? I would like to build something similar. :)

@carstenschroeder
No, I´ve just put it to one B500. This will shut down all other batteries and the EP600. Just add some hotglue to the housing.

![IMG_9678](https://github.com/Patrick762/hassio-bluetti-bt/assets/84518538/cf65c439-143f-401c-b878-5d2176b10305)

**Comment by smai86** [comment]
@Patrick762 
Did you already find some time to watch for main switch button? That feature would be awesome for me.

P.S. I am testing your integration with a AC200Max. It works like a charm. 👍 From time to time my BT Proxy will lose the connection, but thats not a problem for me. after a reboot of HA everything is working fine again

**Comment by Patrick762** [comment]
@smai86 I don't have too much time for this issue, because the other issues are currently a bit more important.
I also didn't receive any answer from bluetti regarding the registers yet.

**Comment by Skydev0h** [comment]
> Should be possible, but I have to check what register the main switch is on. Also, I'd contact Bluetti just to make sure it doesn't affect the device if you turn the EP600 on and off too often (the relays may be a problem)

It seems that the following register is used for the main switch:
`public static final int AC_SWITCH = 2011;`

On the one hand the aforementioned constant is in `ProtocolAddrV2` class.

On the other hand, there are resources:
```
    <string name="device_switch_off_tips_main_switch">Turn off the main switch?</string>
    <string name="device_switch_on_tips_main_switch">Turn on the main switch?</string>

        public static final int device_switch_off_tips_main_switch = 0x7f130577;
        public static final int device_switch_on_tips_main_switch = 0x7f13057a;
```

That are used in an assumed message box to turn on or off main switch:
![image](https://github.com/user-attachments/assets/153a67cb-fafb-4c12-9233-765918cac1fc)

And CtrlAC is parsed from 23th hex character (register 11) in InvBaseSettings that are read from register pack starting on 2000 - that also results in 2011.

Therefore, most likely, it is the register `2011`.
